### PR TITLE
Change cmake branch logic to more explicitly activate crypto extensio…

### PR DESCRIPTION
**Pull request by NanoAkron**
…ns at compile time for each architecture

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -549,39 +549,36 @@ else()
...
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNO_AES")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DNO_AES")
-  elseif(NOT ARM AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X)
-    message(STATUS "AES support enabled")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maes")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -maes")
-  elseif(PPC64LE OR PPC64 OR PPC)
+  if(PPC64LE OR PPC64 OR PPC)
     message(STATUS "AES support not available on POWER")
+    option(NO_AES "Explicitly disable AES support" ${NO_AES})
```
Comment (xiphon) [inline CMakeLists.txt]: How this is intended to work?

**Comment by NanoAkron** [comment]
I think the version of CMake on the armv8 buildbot needs to be updated to work with newer versions of boost

File: CMakeLists.txt [hunk 2]
```diff
@@ -549,39 +549,36 @@ else()
...
     endif()
-  else()
-    message(STATUS "AES support disabled")
+  elseif(NOT ARM AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X)
+    message(STATUS "AES support enabled")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maes")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -maes")
   endif()
 
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_FLAG}")
```
Comment (xiphon) [inline CMakeLists.txt]: Flag should be set explicitly. Don't like this change.

File: CMakeLists.txt [hunk 3]
```diff
@@ -549,39 +549,36 @@ else()
...
       message(STATUS "Crypto extensions enabled for ARMv8")
-      set(ARCH_FLAG "-march=${ARCH}+crypto")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=${ARCH}+aes")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${ARCH}+aes")
     else()
       message(STATUS "Crypto extensions unavailable on your ARMv8 device")
+      option(NO_AES "Explicitly disable AES support" ${NO_AES})
     endif()
-  else()
-    message(STATUS "AES support disabled")
```
Comment (xiphon) [inline CMakeLists.txt]: Don't strip this message log message.

File: CMakeLists.txt [hunk 4]
```diff
@@ -549,39 +549,36 @@ else()
...
+  elseif(NOT ARM AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X)
```
Comment (xiphon) [inline CMakeLists.txt]: No reason to have `AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X` here

Comment (xiphon) [inline CMakeLists.txt]: No reason to have `AND NOT PPC64LE AND NOT PPC64 AND NOT PPC AND NOT S390X` here

**Comment by NanoAkron** [comment]
My mistake. I misinterpreted the syntax of cmake's 'option' function

Will fix and replace.

**Pull request closed without merging**